protocol/error.go

```diff
@@ -26,8 +34,11 @@ const (
 	ErrorCouldNotVerify
 )
 
-// ErrorResponses contains error codes that
-// a response can omit the DirectoryResponse.
+// ErrorResponses contains the server responses indicating that
+// a client request could not be processed due to a
+// malformed client request or an internal server error.
+// Server response messages containing one of these errors therefore
+// omit the DirectoryResponse.
```

Comment:
> Actually ErrorResponses is currently being used in the client side to omit the consistency check

Ok. I didn't realize that this was an explicit purpose of these error responses. I'll definitely make this clear in the doc, thanks!

> it should include `ErrorMalformedDirectoryMessage` as well

Yes, and similarly, `ErrorCouldNotVerify` is another good candidate, I think.
> we should have a better name for this.

Agreed. I was thinking about this earlier and *all* possible responses (except passed and success) start with the "Error-" prefix, but only a handful of them are actually treated as errors. Maybe it makes sense to treat the non-errors more as different results, e.g. change the prefix to something like "Req-" for all request results (including Success?) and "Check-" for all consistency check results? So we'd be dealing with error codes such as "ReqNameNotFound" or "CheckBadCommitment". This also makes it clearer on which end of the protocol each "error" is returned. What do you think?